system/jlib/jlog.cpp

```diff
@@ -1047,19 +1055,33 @@ void RollingFileLogMsgHandler::addToPTree(IPropertyTree * tree) const
     tree->addPropTree("handler", handlerTree);
 }
 
-#define ROLLOVER_PERIOD 86400
-
 void RollingFileLogMsgHandler::checkRollover()
 {
     time_t tNow;
     time(&tNow);
     struct tm ltNow;
     localtime_r(&tNow, &ltNow);
-    if(ltNow.tm_year != startTime.tm_year || ltNow.tm_yday != startTime.tm_yday)
+    if (ltNow.tm_year != startTime.tm_year || ltNow.tm_yday != startTime.tm_yday)
     {
         localtime_r(&tNow, &startTime);  // reset the start time for next rollover check
         doRollover(true);
     }
+    else if (maxLogFileSize)
+    {
+        linesLoggedSinceSizeChecked++;
+        if (linesLoggedSinceSizeChecked > sizeCheckInterval)
+        {
+            long fsize = ftell(handle);
+            if ((fsize==-1 && errno==EOVERFLOW) || (fsize >= maxLogFileSize))
+            {
+                localtime_r(&tNow, &startTime);
+                doRollover(false);
+            }
+            else
+                sizeCheckInterval = (maxLogFileSize - fsize) / 160 + 1;
```

Comment:
I can see that toward the end of the maxLogFileSize you would want to start checking more frequently, but I'm not sure the interval has to shrink each time as we go along.  Depends a bit on the size of each log line I guess.

Reply:
I could do a fixed interval initially and then shrink when it get near the limit.  However, the recalculation is quick and I didn't think it was worth complicating the code.  Do you think it's worth changing this?